lib/github.com/mpllang/mpllib/AugMap.sml

```diff
@@ -182,12 +180,12 @@ struct
     else (f1(), f2())
 
   fun join m1 k v m2 =
-    case T.balance of
-      WB _ => WBST.join m1 k v m2
+    let val WB _ = T.balance
+    in WBST.join m1 k v m2 end
```

Comment:
I prefer the single-branch `case` here because the idea was that `T.balance` could be extended with other balancing criteria in the future.